enhancements/20231108-elemental-cloudinit-crd.md

```diff
@@ -0,0 +1,176 @@
+# Kubernetes-native Elemental cloud-init documents
+
+## Summary
+
+It is both an expected and supported behavior for Harvester users to
+persist certain configurations to their Harvester nodes. These are currently
+done with Elemental cloud-init documents under the `/oem` path on the system.
+
+Currently, Harvester users would have to write these cloud-init documents
+and manually push them to each node where they would like to effect
+the desired change.
+
+There is a cloud-native tooling gap with regards to this process, however.
+Harvester users might prefer to use a GitOps-style approach to configuring
+their Harvester clusters, where Kubernetes objects are version-controlled
+as YAML and each new revision is applied to the cluster.
+
+This HEP presents a new Custom Resource Definition (CRD) that is surfaced
+via the harvester-node-manager, where a Harvester user could apply a "CloudInit"
+CR to a Harvester cluster, like so:
+
+```yaml
```

Comment:
Yes, I will add commentary on the Status to the HEP. I just omitted it from this snippet here specifically because I just wanted to showcase the part of the object that the user must write.